### PR TITLE
[Shaman] Astral Shift damage reduction module

### PR DESCRIPTION
**Pull request by niseko**
https://www.wowhead.com/spell=108271/astral-shift
image
https://www.warcraftlogs.com/reports/vfktbD3mCaNR7Q8M/#fight=69&source=28

### CODE REVIEW

File: src/parser/shaman/shared/spells/AstralShift.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  get totalDrps() {
+    return this.damageReduced / this.owner.fightDuration * 1000;
+  }
+
+  statistic() {
+    const tooltip = `
+      The total estimated damage reduced was ${formatThousands(this.damageReduced)}.<br /><br />
+ 
+      This is the lowest possible value. This value is pretty accurate for this log if you are looking at the actual gain over not having Astral Shift bonus at all, but the gain may end up higher when taking interactions with other damage reductions into account.`;
```
[the remark this answers is not in the document]
Comment (niseko) [reply src/parser/shaman/shared/spells/AstralShift.js]: Its taking the damage taken amount, so the already reduced value to get a low-end estimation regardless of other DRs that are active. Highest would be true if I had used the unmitigated amount.

[the remark this answers is not in the document]
Comment (Gebuz) [reply src/parser/shaman/shared/spells/AstralShift.js]: yeah I realized afterwards. One thing though, shouldn't it be `const damageTaken = event.amount + (event.absorbed || 0);`?

[the remark this answers is not in the document]
Comment (niseko) [reply src/parser/shaman/shared/spells/AstralShift.js]: I suppose - somehow I feel like there's something speaking against that but I can't think of anything.

[the remark this answers is not in the document]
Comment (niseko) [reply src/parser/shaman/shared/spells/AstralShift.js]: Added absorbed damage to this and also the spirit wolf module as I implemented that one the same way.

**Comment by emallson** [comment]
looks like you addressed the comments, and everything else looks good

**Pull request merged**